Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -105,11 +111,12 @@ double MeasurementSelector::calculateChi2(
 
 MeasurementSelector::Cuts MeasurementSelector::getCutsByTheta(
     const InternalCutBins& config, double theta) {
-  const double absTheta = std::abs(theta);
-
   auto it = std::find_if(config.begin(), config.end(),
-                         [absTheta](const InternalCutBin& cuts) {
-                           return absTheta < cuts.maxAbsTheta;
+                         [theta](const InternalCutBin& cuts) {
+                           // since theta is in [0, pi] and we have a symmetric
+                           // cut in eta, we can just look at the positive half
+                           // of the Z axis
+                           return std::min(theta, M_PI - theta) < cuts.maxTheta;
```

Comment:
Good catch. You can move the `std::min(theta, M_PI - theta)` outside the algorithm.

Have you tested the eta-dependent cuts? I realise my tests in `full_chain_itk.py` just have one cut for all eta - and it is the currently same in Athena.

Reply:
just tested it manually with code changes in Python and it seems to work